addons/mail/models/mail_message.py

```diff
@@ -1171,11 +1171,19 @@ def _notify_message_notification_update(self):
                     continue
                 else:
                     messages |= message
-        updates = [[
-            (self._cr.dbname, 'res.partner', author.id),
-            {'type': 'message_notification_update', 'elements': self.env['mail.message'].concat(*author_messages)._message_notification_format()}
-        ] for author, author_messages in groupby(messages.sorted('author_id'), itemgetter('author_id'))]
-        self.env['bus.bus'].sendmany(updates)
+        data_dict = defaultdict(lambda: self.env['mail.message'])
```

Comment:
Name the dict according to what it contains, in this case `messages_per_partner` or something similar.